packages/wallet/test/InMemoryUtxoRepository.test.ts

```diff
@@ -109,64 +150,92 @@ describe('InMemoryUtxoRepository', () => {
       numUtxoPreTransaction = utxoRepository.allUtxos.length;
-      onTransactionUntracked = jest.fn();
-      utxoRepository.on(UtxoRepositoryEvent.TransactionUntracked, onTransactionUntracked);
+      rewardsPreTransaction = utxoRepository.allRewards!;
+      onOutOfSync = jest.fn();
+      utxoRepository.on(UtxoRepositoryEvent.OutOfSync, onOutOfSync);
     });
 
     it('preconditions', () => {
       expect(utxoRepository.availableUtxos).toHaveLength(utxoRepository.allUtxos.length);
       expect(utxoRepository.availableUtxos).toContain(transactionUtxo);
+      expect(utxoRepository.availableRewards).toBe(rewardsPreTransaction);
     });
 
-    it('transaction confirmed', async () => {
-      let completeConfirmation: Function;
-      const confirmed = new Promise<void>((resolve) => (completeConfirmation = resolve));
-      await trackTransaction(confirmed);
+    describe('sync success', () => {
+      beforeEach(() => {
+        // Simulate spent utxo and rewards
+        expect(provider.utxoDelegationAndRewards).toBeCalledTimes(1);
+        provider.utxoDelegationAndRewards.mockResolvedValueOnce({
+          utxo: utxo.slice(1),
+          delegationAndRewards: {
+            ...delegationAndRewards,
+            rewards: rewards - transactionWithdrawal
+          }
+        });
+      });
 
-      // transaction confirmed
-      await completeConfirmation!();
-      expect(utxoRepository.availableUtxos).toHaveLength(numUtxoPreTransaction - 1);
-      expect(utxoRepository.availableUtxos).toHaveLength(utxoRepository.allUtxos.length);
-      expect(utxoRepository.availableUtxos).not.toContain(transactionUtxo);
-    });
+      it('transaction confirmed', async () => {
+        await trackTransaction(new Promise<void>((resolve) => (completeConfirmation = resolve)));
+      });
 
-    it('transaction timed out', async () => {
-      // setup for transaction to timeout
-      let completeConfirmation: Function;
-      const confirmed = new Promise<void>(
-        (_, reject) => (completeConfirmation = () => reject(new TransactionError(TransactionFailure.Timeout)))
-      );
-      await trackTransaction(confirmed);
+      it('transaction confirmation failed', async () => {
+        // setup for transaction to timeout
+        await trackTransaction(
+          new Promise<void>(
+            (_, reject) => (completeConfirmation = () => reject(new TransactionError(TransactionFailure.Timeout)))
+          )
+        );
+      });
 
-      // transaction rejected
-      await completeConfirmation!();
-      expect(onTransactionUntracked).not.toBeCalled();
-      expect(utxoRepository.availableUtxos).toHaveLength(numUtxoPreTransaction);
-      expect(utxoRepository.availableUtxos).toHaveLength(utxoRepository.allUtxos.length);
-      expect(utxoRepository.availableUtxos).toContain(transactionUtxo);
+      afterEach(async () => {
+        // Review: is it ok to reuse code by utilizing afterEach, or should we only use it for test cleanup?
```

Comment:
I will always lean towards clear test structure for best communication over convenience of writing the tests! :smiley: 

Best to make the primary assertions within the test case, so that a developer can easy comprehend the behaviour. I would expect assertions used within pre/post blocks to be solely based on setup or cleanup states.